exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -281,97 +400,82 @@ func (e *Exchange) UpdateOrderbook(ctx context.Context, pair currency.Pair, asse
 	if err := e.CurrencyPairs.IsAssetEnabled(assetType); err != nil {
 		return nil, err
 	}
-	callingBook := &orderbook.Book{
+	fPair, err := e.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
+		return nil, err
+	}
+	book := &orderbook.Book{
 		Exchange:          e.Name,
-		Pair:              pair,
+		Pair:              fPair,
 		Asset:             assetType,
 		ValidateOrderbook: e.ValidateOrderbook,
 	}
-	orderbookNew, err := e.GetOrderbook(ctx, "", poloniexMaxOrderbookDepth)
-	if err != nil {
-		return callingBook, err
-	}
-
-	enabledPairs, err := e.GetEnabledPairs(assetType)
-	if err != nil {
-		return callingBook, err
-	}
-	for i := range enabledPairs {
-		pFmt, err := e.GetPairFormat(assetType, true)
+	switch assetType {
+	case asset.Spot:
+		var orderbookNew *OrderbookData
+		orderbookNew, err = e.GetOrderbook(ctx, fPair, 0, 150)
 		if err != nil {
```

Comment:
🐒 Optional nitpick:
This seems a mixture of reasons to pre-declare and reasons to use `:=`
Should be one or the other
```suggestion
		orderbookNew, err := e.GetOrderbook(ctx, fPair, 0, 150)
		if err != nil {
```